### ISSUE TITLE
Can only Upload one Csv file than the function will no longer accept Csv files [Breaks]

### ISSUE THREAD

**Issue by SudoCory** [JoshClose/CsvHelper#1768]
Hello CsvHelper Team,

I am trying to make an ASP.NET core 5 web app that has the ability to for a CSV file to be uploaded into a Database. The bug only appears when I am hosting the app live on Azure. The first time a CSV file is uploaded it is properly entered into the database, the second time with the exact same file it returns the following error. This occurs forever with any file (good or bad) until the web app is deleted and re-uploaded. 

![image](https://user-images.githubusercontent.com/60627911/114125607-d507c400-98c4-11eb-9114-b6168e5d7029.png)

From this image it seems as though the index of the IReader and IParser states are not being reset properly. Catching the error simply tells me that it failed, and still will not upload the contents of the data into my database.

My code follows the "Reading by hand" method:
![image](https://user-images.githubusercontent.com/60627911/114126074-b229df80-98c5-11eb-843e-d1ed8d326f97.png)

It is incredibly difficult to debug since it only occurs when it is already being hosted. Any point in the right direction would be appreciated deeply as I cannot find anything on Google to fix this bug.

Thanks and have a good one.

- Cory

**Comment by JoshClose** [comment]
Is the `RawRecord` your header?

**Comment by SudoCory** [comment]
I am not super familiar with how to use the CsvHelper yet, thus I am not sure that exactly the **RawRecord** is?

In my code I am receiving the file (is it passed as an IFormFile), saving it in a directory then pulling to read from and create a set of objects. 

If my header is a **RawRecord**, that do I need to do to fix it?

**Comment by JoshClose** [comment]
In the exception message there is a part `RawRecord` that is cut off. I'm wondering what the text is there.

**Comment by SudoCory** [comment]
Here it is:

CsvHelper.ReaderException: No header record was found.

IReader state:

   ColumnCount: 0

   CurrentIndex: -1

   HeaderRecord:



IParser state:

   ByteCount: 0

   CharCount: 0

   Row: 1

   RawRow: 1

   Count: 0

   RawRecord:





   at CsvHelper.CsvReader.ReadHeader()

   at PartsApp.Controllers.PartsController.UploadCSV(IFormFile csvFile) in C:\local\Temp\zipdeploy\extracted\Controllers\PartsController.cs:line 481

   at lambda_method424(Closure , Object , Object[] )

   at Microsoft.AspNetCore.Mvc.Infrastructure.ActionMethodExecutor.SyncActionResultExecutor.Execute(IActionResultTypeMapper mapper, ObjectMethodExecutor executor, Object controller, Object[] arguments)

   at Microsoft.AspNetCore.Mvc.Infrastructure.ControllerActionInvoker.InvokeActionMethodAsync()

   at Microsoft.AspNetCore.Mvc.Infrastructure.ControllerActionInvoker.Next(State& next, Scope& scope, Object& state, Boolean& isCompleted)

   at Microsoft.AspNetCore.Mvc.Infrastructure.ControllerActionInvoker.InvokeNextActionFilterAsync()

--- End of stack trace from previous location ---

   at Microsoft.AspNetCore.Mvc.Infrastructure.ControllerActionInvoker.Rethrow(ActionExecutedContextSealed context)

   at Microsoft.AspNetCore.Mvc.Infrastructure.ControllerActionInvoker.Next(State& next, Scope& scope, Object& state, Boolean& isCompleted)

   at Microsoft.AspNetCore.Mvc.Infrastructure.ControllerActionInvoker.<InvokeInnerFilterAsync>g__Awaited|13_0(ControllerActionInvoker invoker, Task lastTask, State next, Scope scope, Object state, Boolean isCompleted)

   at Microsoft.AspNetCore.Mvc.Infrastructure.ResourceInvoker.<InvokeNextResourceFilter>g__Awaited|24_0(ResourceInvoker invoker, Task lastTask, State next, Scope scope, Object state, Boolean isCompleted)

   at Microsoft.AspNetCore.Mvc.Infrastructure.ResourceInvoker.Rethrow(ResourceExecutedContextSealed context)

   at Microsoft.AspNetCore.Mvc.Infrastructure.ResourceInvoker.Next(State& next, Scope& scope, Object& state, Boolean& isCompleted)

   at Microsoft.AspNetCore.Mvc.Infrastructure.ResourceInvoker.<InvokeFilterPipelineAsync>g__Awaited|19_0(ResourceInvoker invoker, Task lastTask, State next, Scope scope, Object state, Boolean isCompleted)

   at Microsoft.AspNetCore.Mvc.Infrastructure.ResourceInvoker.<InvokeAsync>g__Awaited|17_0(ResourceInvoker invoker, Task task, IDisposable scope)

   at Microsoft.AspNetCore.Routing.EndpointMiddleware.<Invoke>g__AwaitRequestTask|6_0(Endpoint endpoint, Task requestTask, ILogger logger)

   at Microsoft.AspNetCore.Authorization.Policy.AuthorizationMiddlewareResultHandler.HandleAsync(RequestDelegate next, HttpContext context, AuthorizationPolicy policy, PolicyAuthorizationResult authorizeResult)

   at Microsoft.AspNetCore.Authorization.AuthorizationMiddleware.Invoke(HttpContext context)

   at Microsoft.AspNetCore.Authentication.AuthenticationMiddleware.Invoke(HttpContext context)

   at Microsoft.AspNetCore.Diagnostics.DeveloperExceptionPageMiddleware.Invoke(HttpContext context)

**Comment by JoshClose** [comment]
It looks like the `RawRecord` is empty.

Change the first `Read` to

```cs
var hasMore = csv.Read();
csv.ReadHeader();
```

Is `hasMore` `false`?

**Comment by SudoCory** [comment]
In order to check if the hasMore Variable is set of false I had to make a page for it instead of debugging since the error only occurs when the app is being hosted in production. I tested the function again using my CSV file that works when the app is running locally and not on Azure. 

After adding the var hasMore = csv.Read() I get the following unexpected exception:


CsvHelper.TypeConversion.TypeConverterException: The conversion cannot be performed.

    Text: 'Cost($)'

    MemberType: 

    TypeConverter: 'CsvHelper.TypeConversion.DecimalConverter'

IReader state:

   ColumnCount: 0

   CurrentIndex: 6

   HeaderRecord:

["Material","Description","Size(A)","Size(B)","Size(C)","Size(D)","Cost($)","Quantity","Use","Notes","Supplier"]

IParser state:

   ByteCount: 0

   CharCount: 89

   Row: 1

   RawRow: 1

   Count: 11

   RawRecord:

Material,Description,Size(A),Size(B),Size(C),Size(D),Cost($),Quantity,Use,Notes,Supplier




   at CsvHelper.TypeConversion.DefaultTypeConverter.ConvertFromString(String text, IReaderRow row, MemberMapData memberMapData)

   at CsvHelper.TypeConversion.DecimalConverter.ConvertFromString(String text, IReaderRow row, MemberMapData memberMapData)

   at CsvHelper.CsvReader.GetField(Type type, Int32 index, ITypeConverter converter)

   at CsvHelper.CsvReader.GetField[T](Int32 index, ITypeConverter converter)

   at CsvHelper.CsvReader.GetField[T](String name, ITypeConverter converter)

   at CsvHelper.CsvReader.GetField[T](String name)

   at PartsApp.Controllers.PartsController.UploadCSV(IFormFile csvFile) in C:\local\Temp\zipdeploy\extracted\Controllers\PartsController.cs:line 494

   at lambda_method426(Closure , Object , Object[] )

   at Microsoft.AspNetCore.Mvc.Infrastructure.ActionMethodExecutor.SyncActionResultExecutor.Execute(IActionResultTypeMapper mapper, ObjectMethodExecutor executor, Object controller, Object[] arguments)

   at Microsoft.AspNetCore.Mvc.Infrastructure.ControllerActionInvoker.InvokeActionMethodAsync()

   at Microsoft.AspNetCore.Mvc.Infrastructure.ControllerActionInvoker.Next(State& next, Scope& scope, Object& state, Boolean& isCompleted)

   at Microsoft.AspNetCore.Mvc.Infrastructure.ControllerActionInvoker.InvokeNextActionFilterAsync()

--- End of stack trace from previous location ---

   at Microsoft.AspNetCore.Mvc.Infrastructure.ControllerActionInvoker.Rethrow(ActionExecutedContextSealed context)

   at Microsoft.AspNetCore.Mvc.Infrastructure.ControllerActionInvoker.Next(State& next, Scope& scope, Object& state, Boolean& isCompleted)

   at Microsoft.AspNetCore.Mvc.Infrastructure.ControllerActionInvoker.<InvokeInnerFilterAsync>g__Awaited|13_0(ControllerActionInvoker invoker, Task lastTask, State next, Scope scope, Object state, Boolean isCompleted)

   at Microsoft.AspNetCore.Mvc.Infrastructure.ResourceInvoker.<InvokeNextResourceFilter>g__Awaited|24_0(ResourceInvoker invoker, Task lastTask, State next, Scope scope, Object state, Boolean isCompleted)

   at Microsoft.AspNetCore.Mvc.Infrastructure.ResourceInvoker.Rethrow(ResourceExecutedContextSealed context)

   at Microsoft.AspNetCore.Mvc.Infrastructure.ResourceInvoker.Next(State& next, Scope& scope, Object& state, Boolean& isCompleted)

   at Microsoft.AspNetCore.Mvc.Infrastructure.ResourceInvoker.<InvokeFilterPipelineAsync>g__Awaited|19_0(ResourceInvoker invoker, Task lastTask, State next, Scope scope, Object state, Boolean isCompleted)

   at Microsoft.AspNetCore.Mvc.Infrastructure.ResourceInvoker.<InvokeAsync>g__Awaited|17_0(ResourceInvoker invoker, Task task, IDisposable scope)

   at Microsoft.AspNetCore.Routing.EndpointMiddleware.<Invoke>g__AwaitRequestTask|6_0(Endpoint endpoint, Task requestTask, ILogger logger)

   at Microsoft.AspNetCore.Authorization.Policy.AuthorizationMiddlewareResultHandler.HandleAsync(RequestDelegate next, HttpContext context, AuthorizationPolicy policy, PolicyAuthorizationResult authorizeResult)

   at Microsoft.AspNetCore.Authorization.AuthorizationMiddleware.Invoke(HttpContext context)

   at Microsoft.AspNetCore.Authentication.AuthenticationMiddleware.Invoke(HttpContext context)

   at Microsoft.AspNetCore.Diagnostics.DeveloperExceptionPageMiddleware.Invoke(HttpContext context)


The second time I run the function it returns that HasMore is indeed set to false:

![image](https://user-images.githubusercontent.com/60627911/114241379-0d57e280-9957-11eb-971e-7f138a4e69bc.png)

I wish I could debug and step through my program, but it only occurs when the app is being live hosted.

![image](https://user-images.githubusercontent.com/60627911/114241660-83f4e000-9957-11eb-9568-4e45c9376fde.png)
